### DISCUSSION TITLE
how to change variable in a easy way

### DISCUSSION THREAD

**Discussion by gipsyh** [nushell/nushell#5391] [Q&A] [answered]
```
def main [--arg: string] {
    if $arg == null {
        arg = "arg"
    }
}
```
that can't pass compile
```
def main [--arg: string] {
    let arg = if $arg == null {
        "arg"
    } else {
        $arg
    }
}
```
that not elegant

**Comment by kubouch** [comment, marked as answer]
Nushell is currently immutable so there is no way right now. We might add a limited support for mutability so you could change already defined variable, though, so this might improve in the future.

**Comment by DejayRezme** [comment]
There is also a syntax for default parameter values (I didn't know this existed either, this seems to be newly documented)

https://www.nushell.sh/book/custom_commands.html#parameters-with-a-default-value

For more complex scripting logic Nushell still needs variable mutability in my opinion though (see #5284).
